From: Bala-Vignesh-Reddy <reddybalavignesh9979@gmail.com>
Subject: selftests: centralize include path for kselftest.h and kselftest_harness.h
Date: Wed, 27 Aug 2025 20:17:32 +0530

Patch series "selftests: Centralize kselftest headers to avoid relative
includes".

This series centralizes the handling of kselftest.h and
kselftest_harness.h includes in selftests, replacing relative paths with a
non-relative approach using shared -I path.

Patch 1 updates the build files (Makefile and lib.mk) and include CFLAGS
in sync/Makefile to resolve not found errors.

Patch 2 applies bulk source change (patch 2 is large but it is replaced
automatically).


This patch (of 2):

Add compile flag in Makefile and export it in lib.mk.  Also add CFLAGS in
the Makefile rules to fix include path errors for kselftest.h.

Link: https://lore.kernel.org/lkml/20250820143954.33d95635e504e94df01930d0@linux-foundation.org/
Link: https://lkml.kernel.org/r/20250827144733.82277-2-reddybalavignesh9979@gmail.com
Signed-off-by: Bala-Vignesh-Reddy <reddybalavignesh9979@gmail.com>
Suggested-by: Andrew Morton <akpm@linux-foundation.org>
Cc: Shuah Khan <shuah@kernel.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 tools/testing/selftests/Makefile |    4 ++++
 tools/testing/selftests/lib.mk   |    2 ++
 2 files changed, 6 insertions(+)

--- a/tools/testing/selftests/lib.mk~selftests-centralize-include-path-for-kselftesth-and-kselftest_harnessh
+++ a/tools/testing/selftests/lib.mk
@@ -199,6 +199,8 @@ clean: $(if $(TEST_GEN_MODS_DIR),clean_m
 # Build with _GNU_SOURCE by default
 CFLAGS += -D_GNU_SOURCE=
 
+CFLAGS += $(KSFT_INCLUDES)
+
 # Enables to extend CFLAGS and LDFLAGS from command line, e.g.
 # make USERCFLAGS=-Werror USERLDFLAGS=-static
 CFLAGS += $(USERCFLAGS)
--- a/tools/testing/selftests/Makefile~selftests-centralize-include-path-for-kselftesth-and-kselftest_harnessh
+++ a/tools/testing/selftests/Makefile
@@ -194,11 +194,15 @@ else
   DEFAULT_INSTALL_HDR_PATH := 1
 endif
 
+# Include path for kselftest.h
+KSFT_INCLUDES := -I${abs_srctree}/tools/testing/selftests
+
 # Prepare for headers install
 include $(top_srcdir)/scripts/subarch.include
 ARCH           ?= $(SUBARCH)
 export BUILD
 export KHDR_INCLUDES
+export KSFT_INCLUDES
 
 # set default goal to all, so make without a target runs all, even when
 # all isn't the first target in the file.
_
